app/instance-initializers/theme.js

```diff
@@ -0,0 +1,33 @@
+import Ember from 'ember';
+
+export function initialize(instance) {
+  var container = instance.container;
+  var theme     = window.localStorage.getItem('defaultUiTheme');
+  var userTheme = container.lookup('service:user-theme');
+
+  if (theme) {
+
+    if (theme !== 'ui-auto') {
+      Ember.$('link[rel="stylesheet"]').after(`<link rel="stylesheet" href="assets/${theme}.css">`);
+    } else {
+      userTheme.setAutoUpdate();
+    }
+
+  } else {
+
+    theme = 'ui-light';
+
+    window.localStorage.setItem('defaultUiTheme', theme);
```

Comment:
Any reason this uses storage directly instead of the `preferences` service?  It doesn't explode apparently, but this is saving un-JSON-encoded strings that preferences will see and do..something.. when it can't decode